### ISSUE TITLE
PCC Drop in rtdetr-r18vd, PCC=-0.03929151454276051

### ISSUE THREAD

**Issue by kamalrajkannan78** [tenstorrent/tt-xla#3219]
### Describe the bug

- `rtdetr-r18vd`(**red model**) variant having `PCC=-0.03929151454276051`- [feb9_rtdetr_r18.log](https://github.com/user-attachments/files/25181729/feb9_rtdetr_r18.log)
- Model returns 14 outputs. Among them, `init_reference_points` exhibits a significantly low PCC compared to the others.  
- Since logits themselves already have a low PCC, the focus should be on identifying the earliest point where PCC drops below 0.99.
- The PCC degradation first originates from the [less_than](https://github.com/huggingface/transformers/blob/8cb5963cc22174954e7dca2c0a3320b7dc2f4edc/src/transformers/models/rt_detr/modeling_rt_detr.py#L1578C49-L1578C52) op. `PCC=0.815519325543514` - [feb9_sanity.log](https://github.com/user-attachments/files/25181935/feb9_sanity_f_details.log)
- **Note :** output type of the less_than operation changes from i1 (boolean) to bfloat16 in the TTNN graph.

| #  | Output Field                          | PCC      |
|----|--------------------------------------|----------|
| 0  | logits                               | 0.27932  |
| 1  | pred_boxes                           | -0.02515 |
| 2  | last_hidden_state                    | 0.10633  |
| 3  | intermediate_hidden_states           | 0.15499  |
| 4  | intermediate_logits                  | 0.23055  |
| 5  | intermediate_reference_points        | -0.01378 |
| 6  | encoder_last_hidden_state[0]          | 0.99978  |
| 7  | encoder_last_hidden_state[1]          | 0.99983  |
| 8  | encoder_last_hidden_state[2]          | 0.99987  |
| 9  | init_reference_points                | -0.03929 |
| 10 | enc_topk_logits                      | -0.02875 |
| 11 | enc_topk_bboxes                      | -0.02866 |
| 12 | enc_outputs_class                    | 0.98317  |
| 13 | enc_outputs_coord_logits             | 0.89005  |

### Emitpy 

- Able to replicate the drop with emitpy. [feb10_lt_emitpy_run.log](https://github.com/user-attachments/files/25202708/feb10_lt_emitpy_run.log), [main.py](https://github.com/user-attachments/files/25202707/main.py)

### TT-Metal ticket 
- https://github.com/tenstorrent/tt-metal/issues/37471

### Repro

```
git checkout kkannan/feb9_pcc_drop_rtdetr_lt
git submodule update --init —recursive
git lfs pull --include ‘*.pt'

# model run 
pytest tests/runner/test_models.py::test_all_models_torch[rt_detr/pytorch-rtdetr-r18vd-single_device-inference] -svv --runxfail 

# sanity run 
pytest tests/torch/ops/test_sanity.py -svv

# to generate TTNN Repro
python examples/pytorch/codegen/python/less_than.py

# pcc comparison 
# unzip the attached emitpy zip file and place the contents of attached main.py in less_than/main.py 

cd less_than/
./run
```

### logs

- [less_than_emitpy.zip](https://github.com/user-attachments/files/25182033/less_than.zip)
- [feb9_lt_emitpy_gen.log](https://github.com/user-attachments/files/25203032/feb9_lt_emitpy_gen.log)
